server/test/e2e/asset.e2e-spec.ts

```diff
@@ -49,83 +53,498 @@ const makeUploadDto = (options?: { omit: string }): Record<string, any> => {
+        deferred: () => ({
+          query: { takenAfter: yesterday.minus({ hour: 1 }).toJSDate() },
+          assets: [asset3],
+        }),
+      },
+      {
+        should: 'should search by takenAfter (no results)',
+        deferred: () => ({
+          query: { takenAfter: today.plus({ hour: 1 }).toJSDate() },
+          assets: [],
+        }),
+      },
+      {
+        should: 'should search by originalPath',
+        deferred: () => ({
+          query: { originalPath: asset1.originalPath },
+          assets: [asset1],
+        }),
+      },
+      {
+        should: 'should search by originalFilename',
+        deferred: () => ({
+          query: { originalFileName: asset1.originalFileName },
+          assets: [asset1],
+        }),
+      },
+      {
+        should: 'should search by encodedVideoPath',
+        deferred: () => ({
+          query: { encodedVideoPath: '/path/to/encoded-video.mp4' },
+          assets: [asset3],
+        }),
+      },
+      {
+        should: 'should search by resizePath',
+        deferred: () => ({
+          query: { resizePath: '/path/to/thumb.jpg' },
+          assets: [asset3],
+        }),
+      },
+      {
+        should: 'should search by webpPath',
+        deferred: () => ({
+          query: { webpPath: '/path/to/thumb.webp' },
+          assets: [asset3],
+        }),
+      },
+      {
+        should: 'should search by city',
+        deferred: () => ({
+          query: { city: 'Immich' },
+          assets: [asset3],
+        }),
+      },
+      {
+        should: 'should search by state',
+        deferred: () => ({
+          query: { state: 'Nebraska' },
+          assets: [asset3],
+        }),
+      },
+      {
+        should: 'should search by country',
+        deferred: () => ({
+          query: { country: 'United States' },
+          assets: [asset3],
+        }),
+      },
+      {
+        should: 'sohuld search by make',
+        deferred: () => ({
+          query: { make: 'Cannon' },
+          assets: [asset3],
+        }),
+      },
+      {
+        should: 'should search by country',
+        deferred: () => ({
+          query: { model: 'EOS Rebel T7' },
+          assets: [asset3],
+        }),
+      },
+      {
+        should: 'should search by lensModel',
+        deferred: () => ({
+          query: { lensModel: 'Fancy lens' },
+          assets: [asset3],
+        }),
+      },
+    ];
+
+    for (const { should, deferred } of searchTests) {
+      it(should, async () => {
+        const { assets, query } = deferred();
+        const { status, body } = await request(server)
+          .get('/assets')
+          .query(query)
+          .set('Authorization', `Bearer ${user1.accessToken}`);
+
+        expect(status).toBe(200);
+        expect(body.length).toBe(assets.length);
+        for (let i = 0; i < assets.length; i++) {
+          expect(body[i]).toEqual(expect.objectContaining({ id: assets[i].id }));
+        }
```

Comment:
Why not just map assets to `[expect.objectContaining({id: ...}), expect.objectContaining({id: ...}), ...]`? And then `expect(body).toEqual(map)`?

Reply:
This also checks the order and makes the diff easier to read when there is a mismatch.

Reply:
The order should be checked by my approach as well (not using `expect.arrayContaining`). The diff argument makes sense though